lib/widgets/details_poster_section.dart

```diff
@@ -0,0 +1,102 @@
+import 'package:flutter/material.dart';
+import '../../../widgets/like_counter.dart';
+import '../../../utils/UIConstants.dart';
+
+class DetailsPosterSection extends StatelessWidget {
+  const DetailsPosterSection({
+    super.key,
+    required this.pathToPosterImg,
+    required this.voteAverage,
+  });
+
+  final String pathToPosterImg;
+  final double voteAverage;
+
+  @override
+  Widget build(BuildContext context) {
+    return SizedBox(
+      width: double.infinity,
+      height: UIConstants.posterSectionSize,
+      child: Row(
+        mainAxisAlignment: MainAxisAlignment.spaceAround,
+        children: [
+          _MoviePoster(
+            pathToPosterImg: pathToPosterImg,
+          ),
+          Column(
+            mainAxisAlignment: MainAxisAlignment.spaceEvenly,
+            children: [
+              _MovieScore(
+                voteAverage: voteAverage,
+              ),
+              const LikeCounter(),
+            ],
+          ),
+        ],
+      ),
+    );
+  }
+}
+
+class _MoviePoster extends StatelessWidget {
+  const _MoviePoster({
+    required this.pathToPosterImg,
+  });
+
+  final String pathToPosterImg;
+  static const double width = 200;
+  static const double height = 300;
+  static const double borderWidth = 2;
+
+  @override
+  Widget build(BuildContext context) {
+    return SizedBox(
+      width: width,
+      height: height,
+      child: Container(
+        decoration: BoxDecoration(
+          border: Border.all(width: borderWidth),
+          image: DecorationImage(
+            image: AssetImage(pathToPosterImg),
+            fit: BoxFit.fill,
+          ),
+        ),
+      ),
+    );
+  }
+}
+
+class _MovieScore extends StatelessWidget {
+  const _MovieScore({
+    required this.voteAverage,
+  });
+
+  final double voteAverage;
+  static const double iconSize = UIConstants.posterSectionIconSize;
+  static const TextStyle scoreTextStyle = TextStyle(
+    color: Colors.white,
+    fontSize: 70,
+    fontWeight: FontWeight.bold,
+    fontStyle: FontStyle.italic,
+  );
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      height: iconSize,
+      width: iconSize,
+      decoration: const BoxDecoration(
+        color: Colors.amber,
+        shape: BoxShape.circle,
+      ),
+      child: Align(
+        alignment: Alignment.center,
+        child: Text(
+          textAlign: TextAlign.center,
+          (voteAverage * 10).toInt().toString(),
```

Comment:
Maybe you can made this validation inside Movie class, you can add a get method like this
```
class Movie {
    String get voteAverage => (this.voteAverage * 10).toInt().toString()
}
```

Reply:
it does make it a lot cleaner, thanks!